File: autozig-math/zig/uvec2.zig
const std = @import("std");

pub const UVec2 = extern struct {
    x: u32,
    y: u32,

    pub fn new(x: u32, y: u32) UVec2 {
        return .{
            .x = x,
            .y = y,
        };
    }
};

export fn uvec2_new(x: u32, y: u32) UVec2 {
    return UVec2.new(x, y);
}

File: autozig-math/src/uvec2.rs
use autozig::include_zig;

#[repr(C)]
#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
pub struct UVec2 {
    pub x: u32,
    pub y: u32,
}

include_zig!("zig/uvec2.zig", {
    fn uvec2_new(x: u32, y: u32) -> UVec2;
});

impl UVec2 {
    pub const ZERO: Self = Self { x: 0, y: 0 };
    pub const ONE: Self = Self { x: 1, y: 1 };
    pub const X: Self = Self { x: 1, y: 0 };
    pub const Y: Self = Self { x: 0, y: 1 };

    pub fn new(x: u32, y: u32) -> Self {
        uvec2_new(x, y)
    }
}
